src/NSubstitute/Proxies/CastleDynamicProxy/CastleInvocationMapper.cs

```diff
@@ -10,8 +10,7 @@ public virtual ICall Map(IInvocation castleInvocation)
         Func<object>? baseMethod = null;
         if (castleInvocation.InvocationTarget != null &&
             castleInvocation.MethodInvocationTarget.IsVirtual &&
-            !castleInvocation.MethodInvocationTarget.IsAbstract &&
-            !castleInvocation.MethodInvocationTarget.IsFinal)
+            !castleInvocation.MethodInvocationTarget.IsAbstract)
```

Comment:
**question:** dropping "not final" constraint here is required to support call forwarding?